src/TLabel.h

```diff
@@ -68,6 +68,7 @@ class TLabel : public QLabel
     int mWheelFunction = 0;
     int mEnterFunction = 0;
     int mLeaveFunction = 0;
+    QMovie* myMovie = nullptr;
```

Comment:
Apropos the comment about memory leaks you need a destructor for the `TLabel` class now, probably something of the form:
```cpp
    ~TLabel()
    {
        if (myMovie) {
            myMovie->deleteLater();
            myMovie = nullptr;
        }
    }
````
also `myMovie` is probable not the best identifier for it - how about `mpMovie` instead - `m` indicating a class member and `p` that it is a pointer...

Reply:
Strangely, measing the actual memory use - this does not seem to be so necessary. But it can't hurt...

Reply:
It shouldn't be necessary because I use setParent on the QMovie to make it a child of the Label so when the label is deleted it should delete also his children. I will add it anyway to be 100% sure.